projects/canopy/src/lib/variant/variant.directive.ts

```diff
@@ -0,0 +1,24 @@
+import { Directive, ElementRef, Input, Renderer2 } from '@angular/core';
+
+import { Variant } from './variant.interface';
+
+@Directive({
+  selector: '[lgVariant]',
+})
+export class LgVariantDirective {
+  variantClass: string;
+  @Input()
+  set lgVariant(variant: Variant) {
+    this.variantClass = this.toggleClass(`lg-variant--${variant}`, this.variantClass);
+  }
+
+  toggleClass(newClass: string, oldClass: string): string {
```

Comment:
Does this need to be a function if it is only ever called from the input setter? I only ask as something about the name `toggleClass` and the fact it also returns the class name doesn't seem quite right. Sounds a bit nitpicky I know!

Reply:
Good point, definitely can see what you mean. Actually this function is copied from the `lgMargin` and `lgPadding` directives, which do use the function multiple times throughout the file. I was actually thinking that at some point, we should create a util method or service for this, but I wanted to open a discussion for that and agree on an approach, as we're going to be adding more of these util directives. So in this case, I thought just keeping the functionality the same as `lgMargin` and `lgPadding` would mean it would be easier to apply some common logic later.

Reply:
I think in this new version of Angular we can avoid doing this bit of code. I think there is an issue (or there was in canopy collective) to go through the codebase and update it.

Reply:
Ah cool, is there a new feature in Angular that does this bit of a logic for us?

Reply:
Yes, but I can't remember what it is :D